src/clj/puppetlabs/services/ca/certificate_authority_core.clj

```diff
@@ -257,6 +257,22 @@
           request-cert)
         (log/warn (i18n/trs "Request is missing a certificate for an endpoint that requires a certificate."))))))
 
+(schema/defn handle-bulk-cert-signing
+  [_request
+   ca_settings :- ca/CaSettings]
```

Comment:
In this case, you want to get the `certnames` out of the `body` of the request. See https://github.com/puppetlabs/puppetserver/pull/2799/files#diff-caf7efdcbdce3e60f52b15dd0489e7f16cad9c1dcfe2993ef1099ec843a50db8L177 for an example.   You can validate that the body is a schema like:
```
(schema/def Certnames [schema/Str])
```
The validation will have to be done manually.  I'll see if I can find an example.

Reply:
I didn't find a good example.  Basically you just need to call:
`(schema/validate Certnames <parsed body>)` and it will throw an exception which will be caught and handled if it fails.